src/cpp/transport/TCPTransportInterface.cpp

```diff
@@ -888,304 +768,198 @@ bool TCPTransportInterface::read_body(
-                close_tcp_socket(p_channel_resource);
-            }
-            success = false;
-        }
-        catch (const asio::system_error& error)
-        {
-            (void)error;
-            // Close the channel
-            logError(RTCP_MSG_IN, "ASIO SYSTEM_ERROR [RECEIVE]: " << error.what());
-            //p_channel_resource->ConnectionLost();
-            close_tcp_socket(p_channel_resource);
-            success = false;
-        }
     }
-
-    success = success && receive_buffer_size > 0;
-
-    return success;
-}
-/*
-size_t TCPTransportInterface::send(
-        TCPChannelResource* p_channel_resource,
-        const octet* data,
-        size_t size, eSocketErrorCodes& errorCode) const
-{
-    size_t bytesSent = 0;
-    try
-    {
-        asio::error_code ec;
-        std::unique_lock<std::recursive_mutex> scopedLock(p_channel_resource->write_mutex());
-        //bytesSent = p_channel_resource->socket()->send(asio::buffer(data, size), 0, ec);
-        if (p_channel_resource->alive())
-        {
-            bytesSent = p_channel_resource->send(data, size, ec);
-            errorCode = eSocketErrorCodes::eNoError;
-        }
-        else
-        {
-            errorCode = eSocketErrorCodes::eBrokenPipe;
-        }
-
-    }
-    catch (const asio::error_code& error)
+    catch (const asio::error_code& code)
     {
-        logInfo(RTCP, "ASIO [SEND]: " << error.message());
-        if ((asio::error::eof == error.value()) || (asio::error::connection_reset == error.value()))
+        if ((code == asio::error::eof) || (code == asio::error::connection_reset))
         {
-            errorCode = eSocketErrorCodes::eBrokenPipe;
-        }
-        else
-        {
-            errorCode = eSocketErrorCodes::eAsioError;
+            // Close the channel
+            logError(RTCP_MSG_IN, "ASIO [RECEIVE]: " << code.message());
+            //channel->ConnectionLost();
+            close_tcp_socket(channel);
+            channel.reset();
         }
+        success = false;
     }
     catch (const asio::system_error& error)
     {
         (void)error;
-        logInfo(RTCP, "ASIO [SEND]: " << error.what());
-        errorCode = eSocketErrorCodes::eSystemError;
-    }
-    catch (const std::exception& error)
-    {
-        (void)error;
-        logInfo(RTCP, "ASIO [SEND]: " << error.what());
-        errorCode = eSocketErrorCodes::eException;
+        // Close the channel
+        logError(RTCP_MSG_IN, "ASIO SYSTEM_ERROR [RECEIVE]: " << error.what());
+        //channel->ConnectionLost();
+        close_tcp_socket(channel);
+        channel.reset();
+        success = false;
     }
 
-    return bytesSent;
-}
+    success = success && receive_buffer_size > 0;
 
-size_t TCPTransportInterface::send(
-        TCPChannelResource *p_channel_resource,
-        const octet *data,
-        size_t size) const
-{
-    eSocketErrorCodes error;
-    return send(p_channel_resource, data, size, error);
+    return success;
```

Comment:
Removing the line from above:
```suggestion
    return success && receive_buffer_size > 0;
```